packages/ui/src/primitives/tailwind/Modal/index.tsx

```diff
@@ -43,26 +43,41 @@ export interface ModalProps {
   submitButtonDisabled?: boolean
   closeButtonText?: string
   submitButtonText?: string
+  startSubmitIcon?: ReactNode
+  actionsOnHeader?: boolean
   onClose?: (isHeader: boolean) => void
   onSubmit?: () => void
 }
 
 export const ModalHeader = ({
   title,
+  actionsOnHeader,
+  startSubmitIcon,
+  submitButtonText,
+  onSubmit,
   onClose
 }: {
   closeIcon?: boolean
   title?: string
+  startSubmitIcon?: ReactNode
+  actionsOnHeader?: boolean
+  submitButtonText?: string
+  onSubmit?: () => void
```

Comment:
we are adding too many props to the header component which is an anti-pattern
in these cases, we should create our own derived component for the specific use case
For example, the ui/src/components/editor folder does this derivations from the primitives for its components.

Reply:
yes maybe in this case should work better using a new variant, i don't think so we need to create another component because how the primitives are build doesn't have very much extensibility probably the primitives should be headless and have separation of concerns using named children pattern to achieve that extensibility

Reply:
The `Modal` is a high level component and made to be configurable but not extendable (since there can be entirely different ways it can be implemented). This [is a one-off case](https://www.figma.com/design/Ds2NEvvz9nqddltP3lsKoP/Design-Changes?node-id=0-1&t=YLp6WBatFXncmfY7-0) which can be put in the multitenancy project with a specific modal title or in this file with a specific name like `HeaderActionsModal`.